### PR TITLE
Feature/add leads page

### PR DESCRIPTION
**Pull request by rudenkoserhii**
## Describe your changes

- I made a page with list of the Leads (directly and from property).
- Changed fetching from swr to RTK.
- Implemented on property card image slider on click modal window with magnifying glass effect
- Connected Filter component from @anette1983 

Screenshot 2023-12-20 092113
Screenshot 2023-12-24 143535

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/browse/AG-125?atlOrigin=eyJpIjoiYTEzYmM3ZWU5ZjEyNDVjYWJiMzlhNmY1OWFlNjlhMTEiLCJwIjoiaiJ9)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [ ] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by anette1983 — APPROVED** [review]

File: src/app/[locale]/leads/lib/components/button-add-new-lead/button-add-new-lead.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { useTranslations } from 'next-intl';
+
+import { Button } from '@mui/material';
+
+import Iconify from 'components/iconify';
+import { endpoints } from 'utils/axios';
+
+const URL = endpoints.lead;
+
+function ButtonAddNewLead(): JSX.Element {
```
Comment (lyudad) [inline src/app/[locale]/leads/lib/components/button-add-new-lead/button-add-new-lead.tsx]: а чего что то реюзабл не сделать, не улучшить текущие кнопки через пропсы. а то кто будет помнить о кнопках которые зашиты где то в подпапках подпапок

Comment (rudenkoserhii) [reply src/app/[locale]/leads/lib/components/button-add-new-lead/button-add-new-lead.tsx]: done

**Review by anette1983 — APPROVED** [review]

File: src/api/lead.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import { endpoints, fetcherQobrix } from 'utils/axios';
+import { ILead, ILeadsFull, ILeadsParams } from 'types/lead';
+
+const URL = endpoints.lead;
```
Comment (lyudad) [inline src/api/lead.ts:6]: why it needs as separate constant, it used only once

File: src/app/[locale]/global.css [hunk 3]
```diff
@@ -24,16 +24,26 @@ img {
...
+  cursor: pointer;
 }
 
 /* End Normalize */
 
 .slick-dots {
   top: 10rem;
   transform: scale(2);
-  filter: invert(1);
+  /* filter: invert(1); */
```
Comment (lyudad) [inline src/app/[locale]/global.css:36]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.10

File: src/app/[locale]/leads/lib/components/common/common.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import { Button, Box, Typography } from '@mui/material';
+
+import { ProtectedRoute } from 'components/custom';
+import LeadsList from 'components/custom/leadsList';
+
+import Iconify from 'components/iconify';
+import { endpoints } from 'utils/axios';
+import { LeadsFilter } from '../..';
+
+const URL = endpoints.lead;
```
Comment (lyudad) [inline src/app/[locale]/leads/lib/components/common/common.tsx:18]: все еще не понимаю зачем константа, так еще и если она повторяется то зачем 2 константы

File: src/components/custom/leadsList/components/lead-item/lead-item.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+import { useTranslations } from 'next-intl';
+
+import { Box, Card } from '@mui/material';
+import { ILead } from 'types/lead';
+import { QobrixLeadItem } from 'types';
+import { endpoints } from 'utils/axios';
+
+import { TextStyled } from '../../styles';
+
+const URL = endpoints.lead;
```
Comment (lyudad) [inline src/components/custom/leadsList/components/lead-item/lead-item.tsx:14]: тоже самое. это же ухудшает читабельность и генерирует не нужный текст

File: src/app/[locale]/leads/lib/components/common/common.tsx (hunk 4 above)
Comment (rudenkoserhii) [reply src/app/[locale]/leads/lib/components/common/common.tsx:18]: Already fixed in next PR

File: src/components/custom/leadsList/components/lead-item/lead-item.tsx (hunk 5 above)
Comment (rudenkoserhii) [reply src/components/custom/leadsList/components/lead-item/lead-item.tsx:14]: Already fixed in next PR

File: src/app/[locale]/global.css (hunk 3 above)
Comment (rudenkoserhii) [reply src/app/[locale]/global.css:36]: Already fixed in next PR

**Pull request merged**